tests/test_model_tool.py

```diff
@@ -0,0 +1,158 @@
+# This file is part of CycloneDX Python Lib
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) OWASP Foundation. All Rights Reserved.
+
+
+from json import loads as json_loads
+from os.path import join
+from unittest import TestCase
+
+from sortedcontainers import SortedSet
+
+from cyclonedx.exception.model import MutuallyExclusivePropertiesException
+from cyclonedx.model.bom import Bom
+from cyclonedx.model.component import Component
+from cyclonedx.model.contact import OrganizationalEntity
+from cyclonedx.model.service import Service
+from cyclonedx.model.tool import Tool, ToolsRepository, ToolsRepositoryHelper
+from cyclonedx.output.json import JsonV1Dot5
+from cyclonedx.output.xml import XmlV1Dot5
+from tests import OWN_DATA_DIRECTORY, reorder
+from tests._data.models import get_bom_with_tools_with_component_and_service
+
+
+class TestModelTool(TestCase):
+
+    def test_sort(self) -> None:
+        # expected sort order: (vendor, name, version)
+        expected_order = [0, 1, 2, 3, 4, 5, 6]
+        tools = [
+            Tool(vendor='a', name='a', version='1.0.0'),
+            Tool(vendor='a', name='a', version='2.0.0'),
+            Tool(vendor='a', name='b', version='1.0.0'),
+            Tool(vendor='a', name='b'),
+            Tool(vendor='b', name='a'),
+            Tool(vendor='b', name='b', version='1.0.0'),
+            Tool(name='b'),
+        ]
+        sorted_tools = sorted(tools)
+        expected_tools = reorder(tools, expected_order)
+        self.assertListEqual(sorted_tools, expected_tools)
+
```

Comment:
nearly all below this comment  has nothing to to with the `Tool` model.
therefore, it belongs in another test case, not in `TestModelTool`